html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+Also, you should keep your animations to only affecting `opacity` and `transform` if you want absolute best performance for animations on your web page. Yes, our first example above only carried out a simple `background-color` change, but even that was an expensive operation in itself. Have a read of this web.dev article for some handy hints on how to [create high-performance CSS animations](https://web.dev/animations-guide/). It also has a link to [CSS Triggers](https://csstriggers.com/) which will help you see how a CSS property will affect other elements on the page if it is animated. Look at the difference between the `background-color` and `transform` properties.
+
+Like we said earlier, optimization is a rabbit-hole so don't get too bogged down in the details here. What's important is that you have a solid understanding of these concepts and can apply them when necessary because hey, if you need to turn a button into a rainbow when it's hovered over, you better be able to transition that rainbow!
+
+### Keyframe Animations
+Keyframe animations let you animate elements from one style configuration to another. Does this sound familiar? You wouldn't be wrong for thinking, "Well, what's the point in learning keyframes if they are basically the same as transitions?", but there are some aspects of keyframes that make them better suited to plenty of situations.
+
+CSS transitions were designed to animate an element from one state to another. They *can* loop but they weren't designed for that, whereas keyframes were designed with the purpose of explicitly enabling loops.
+
+Transitions need a trigger (remember the pseudo-classes?) but keyframes do not. Once you have your elements in place and CSS defined, an animation will start running immediately  if that's what you told it to do. 
+
+Transitions are also not as flexible as using keyframes. When you define a transition, imagine you are sending that element on a journey in a straight line from point A to point B. Yes the `transition-timing-function` can add some variance to the timing of this change, but it doesn't compare to the amount of flexibility added by using keyframes.
+
+There are more differences between the methods than those described above, but hopefully you have a clearer idea of the differences between them now. If you need to change the opacity of an element when it is active, then a keyframe animation would be overkill. But if you need to carry out something more complicated, keyframes provide you with the tools you need. 
```

Comment:
I'd save this paragraph until after the example... i.e. they probably _don't_ have a clearer idea yet... and the usage tips will make more sense later too.